### PR TITLE
KFP CLI

### PR DESCRIPTION
**Pull request by hongye-sun**
The PR includes 3 commands:
- run: start a run
- list-runs: list recent runs
- get-run: get a run

The CLI tool uses argo CLI to display workflow details in watch mode. 

Screencast: https://drive.google.com/file/d/1QpVD4lLaNCKWXk00afpmPvIBZZeIQB2q/view?usp=sharing

This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1449)

### CODE REVIEW

File: sdk/python/setup.py [hunk 1]
```diff
@@ -67,4 +69,6 @@
     ],
     python_requires='>=3.5.3',
     include_package_data=True,
-    entry_points={'console_scripts': ['dsl-compile = kfp.compiler.main:main',]})
+    entry_points={'console_scripts': [
+        'dsl-compile = kfp.compiler.main:main',
```
Comment (gaoning777) [inline sdk/python/setup.py]: What do you think of exposing one command line tool for both compilation and client submission?
for example: kfp compile(=dsl-compile). kfp client(=kfp)

Comment (hongye-sun) [reply sdk/python/setup.py]: I am going to add "kfp compile" as the next PR and "kfp run" can also support to start a run directly from python file.

File: sdk/python/kfp/__main__.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import sys
+import subprocess
+import pprint
+import time
+import json
+import click
+
+from tabulate import tabulate
+
+@click.group()
```
Comment (gaoning777) [inline sdk/python/kfp/__main__.py]: Maybe add a todo to expose other client APIs such as list_experiment, etc.?

File: sdk/python/setup.py (hunk 1 above)
Comment (gaoning777) [reply sdk/python/setup.py]: Perfect. 
Does that mean the dsl-compile tool will be deprecated, then?

File: sdk/python/kfp/__main__.py (hunk 2 above)
Comment (hongye-sun) [reply sdk/python/kfp/__main__.py]: sg, will add

File: sdk/python/setup.py (hunk 1 above)
Comment (hongye-sun) [reply sdk/python/setup.py]: I think so. I will add deprecation warning in the dsl-compile after adding kfp compile.

Comment (gaoning777) [reply sdk/python/setup.py]: SG

**Comment by gaoning777** [comment]
We need to have some tests to cover the CLI tool. Synced offline, will update the basic e2e tests to use the cli instead of the SDK.

**Pull request merged**